pkg/volume/iscsi/iscsi_util.go

```diff
@@ -897,3 +880,50 @@ func cloneIface(b iscsiDiskMounter) error {
 	}
 	return lastErr
 }
+
+func isSessionBusy(host volume.VolumeHost, portal, iqn string) bool {
+	fsDir := host.GetPluginDir(iscsiPluginName)
+	countFS, err := getVolCount(fsDir, portal, iqn)
+	if err != nil {
+		klog.Errorf("iscsi: could not determine FS volumes in use: %v", err)
+		return true
+	}
+
+	blockDir := host.GetVolumeDevicePluginDir(iscsiPluginName)
+	countBlock, err := getVolCount(blockDir, portal, iqn)
+	if err != nil {
+		klog.Errorf("iscsi: could not determine block volumes in use: %v", err)
+		return true
+	}
+
+	return countFS+countBlock > 1
+}
+
+func getVolCount(dir, portal, iqn string) (int, error) {
+	contents, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return 0, err
+	}
+
+	var counter int
+	for _, c := range contents {
+		if !c.IsDir() || c.Name() == config.DefaultKubeletVolumeDevicesDirName {
+			continue
+		}
+
+		mounts, err := ioutil.ReadDir(filepath.Join(dir, c.Name()))
+		if err != nil {
+			return 0, err
+		}
+
+		for _, m := range mounts {
+			volumeMount := m.Name()
+			prefix := portal + "-" + iqn
+			if strings.HasPrefix(volumeMount, prefix) {
+				counter++
```

Comment:
is this relying on lun to be the ref counter?  That won't work in a RWM case for a block device I don't think

Reply:
Not sure if I understand the question correctly, but the prefix doesn't have the lun, so the counter doesn't rely on it (only on portal + iqn).

Reply:
Yeah, I was worried there was a uniqueness check based on lun number somewhere else in the changes.  Thanks for clarifying.